akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -204,6 +230,35 @@ object ByteString {
 
     private[akka] def byteStringCompanion = ByteString1
 
+    override def dropRight(n: Int): ByteString =
+      dropRight1(n)
+
+    /** INTERNAL API */
+    private[akka] def dropRight1(n: Int): ByteString1 =
```

Comment:
to not hide the fact that it's a `ByteString1` which is statically needed by `ByteStrings` (it contains `Vector[ByteString1]`. So instead of casting blindly, we can use this safe method